.github/workflows/ci.yml

```diff
@@ -59,15 +59,52 @@ jobs:
       run: |
         cabal configure --enable-tests --enable-benchmarks --test-show-details=direct
 
+    - name: Update
+      run: |
+        cabal update
+
     - name: Freeze
       run: |
         cabal freeze
 
+    - name: Setup cabal.project.freeze.cache.key
+      shell: bash
+      run: |
+          # Create a new caba.project.freeze.cache.key file for the cache key, removing the cabal index timestamp in the last line
+          source_file="cabal.project.freeze"
+          target_file="cabal.project.freeze.cache.key"
+
+          # Check if the source file exists
+          if [ ! -e "$source_file" ]; then
+              echo "Source file '$source_file' does not exist."
+              exit 1
+          fi
+
+          # Copy the source file to the target file
+          cp "$source_file" "$target_file"
+
+          # Check if the copy was successful
+          if [ $? -ne 0 ]; then
+              echo "Failed to copy the source file to the target file."
+              exit 1
+          fi
+
+          # Remove the last line from the target file
+          sed -i'' -e '$d' "$target_file"
+
+          # Check if the removal of the last line was successful
+          if [ $? -ne 0 ]; then
+              echo "Failed to remove the last line from the target file."
+              exit 1
+          fi
+
+          echo "File '$target_file' created with the last line removed."
+
```

Comment:
I recommend simply `cabal build --dry-run` and then use the hash of `dist-newstyle/cache/plan.json` as the cache key: https://github.com/haskell-actions/setup#model-cabal-workflow-with-caching

Reply:
That's definitely more simple. I suppose that settles it @tomjaguarpaw ?

Reply:
Sounds great to me!